### ISSUE TITLE
Traefik labels documentation ( docker swarm mode )

### ISSUE THREAD

**Issue by Xustyx** [containous/traefik#1290]
### What version of Traefik are you using (`traefik version`)?
Last.

### What is your environment & configuration (arguments, toml...)?
Docker swarm mode with default conf.

### What did you do?
Read all doc ( https://docs.traefik.io/ )

### What did you expect to see?
A full **Label** documentation.

### What did you see instead?
A partial **Label** documentation.

Hi,

I'm working with docker swarm mode and I want to use traefik to proxy my services.
I'm used to use labels for generate automatically configs and other stuff, but your documentation about how to use trafik with labels could be better and show all traefik potencial.

Thanks

**Comment by emilevauge** [comment]
@Xustyx Could give us more details? Which information is lacking from the documentation?

**Comment by Xustyx** [comment]
For example:
traefik.frontend.rule=Host:test.traefik.io: override the default frontend rule (Default: Host:{containerName}.{domain}).

What about other rules?
Why this info is on https://docs.traefik.io/toml/#docker-backend? First, I gone to the https://docs.traefik.io/basics/ and expect to found labels afeter every "Concept". After, I gone to the https://docs.traefik.io/user-guide/swarm-mode/ and expect the same.

**Comment by robbert229** [comment]
I also found the documentation to be lacking.

**Comment by Routhinator** [comment]
Agreed. In my case I am attempting to determine what label will enable swarm mode so I don't need to alter the command flags from docker-compose. Surely it must exist?

**Comment by zpydee** [comment]
I must agree that I also found documentation in general lacking.

As an app developer with limited reverse proxy and advanced docker networking knowledge, I am finding integrating traefik into a multi-app swarm mode environment to be really challenging. I would suggest building on scenarios where there is a single traefik network for proxying incoming calls together with multiple internal networks for the various app stacks.

Also, I think some documentation on the use of various traefik flags would be really useful so one can understand better exactly why one would use each flag to suit requirements.

**Comment by timoreimann** [comment]
@zpydee 

>Also, I think some documentation on the use of various traefik flags would be really useful so one can understand better exactly why one would use each flag to suit requirements.

Feel free to file dedicated issues if you feel that a particular flag (or group of flags) isn't clear as this issue is about Docker swarm labels.

**Comment by zpydee** [comment]
ok, so in that can what does the flag --docker-swarmmode do exactly (for example), because I have a traefik proxy running with a swarm without that flag and my proxy fails if I add it? My point being that docs for flags across the board (including those for swarm mode) could be improved.

**Comment by timoreimann** [comment]
@zpydee I'm not too familiar with Docker Swarm but my guess is that you need that flag in order to enable Docker Swarm mode [as shown in the user guide](https://docs.traefik.io/user-guide/swarm-mode/).

It's difficult for us to improve documentation if we don't know what parts are exactly hard to understand. We could try to improve across the board but that might not meet your needs.

A more specific list of hard-to-understand flags would be super helpful and greatly appreciated.

**Comment by esconsut1** [comment]
I think what's missing is better examples in say, docker-compose.yml file
or
this-label: means that
that-other-label: does that other thing
and-this-label: it makes this happen

All that doc is missing.
Maybe its there, but as others have said, hard to find.

**Comment by elfrinjo** [comment]
Another example:
I am trying to X-Frame-Options, in the configuration-file there is an option "FrameDeny=true" which does what I want.
I guess, this is also possible with labels - but I can not find any example or reference to the correct syntax.

**Comment by emmetog** [comment]
Same here, I'm interested in setting up letsencrypt via labels as opposed to mapping in a config.toml file. I can't find the exact labels that I should be using to set up letsencrypt.

Is there a "standard" way that labels map to config options? For example, I see [here](https://gist.github.com/nknapp/20c7cd89f1f128b8425dd89cbad0b802) the relevant section that I need if I were mounting the config into the container:
```toml
# Enable ACME (Let's Encrypt): automatic SSL
[acme]
# caServer = "https://acme-staging.api.letsencrypt.org/directory"
email = "letsencrypt@example.com"
storage = "acme.json" # or "traefik/acme/account" if using KV store
entryPoint = "https"
onDemand = false
OnHostRule = true
```

Are the labels then? :
* traefik.acme.email
* traefik.acme.storage
* ...

Basically the way that the labels map to the config file could do with being better documented.

**Comment by traefiker** [comment]
I'll close the issue due to inactivity but feel free to re-open if needed.

**Comment by else** [comment]
This is still relevant.

**Comment by ldez** [comment]
Now, all available labels are in the documentation : https://docs.traefik.io/configuration/backends/docker/#labels-overriding-default-behaviour

**Comment by datacarl** [comment]
Hi!

First of all, thanks for making traefik! It's soooo good with docker!

I'd love to see docs on how to set headers based on a path when running in swarm mode. I.e., in my compose file I have a service with the name webapp. For some paths in the webapp, but not all, I want to add a custom response header. If I was defining the frontends in the toml file I'd just add another frontend with a matcher, but I can't find any docs on how to add more frontends for a single backend in swarm mode.

I tried to set a label with a matcher, like `traefik.frontend.headers.customRequestHeaders=<EXPR>; Path: /some/path`, but that just makes the `; Path....` part part of the header.

Maybe what I'm trying to do doesn't work in swarm mode. If that's the case it would be nice if that was made clear somewhere.

(For the record, although this is somewhat unrelated to this issue, I also tried to configure a frontend using the toml file, although I'm running in swarm mode, E.g. 

```
[frontends]
  [frontends.whoami]
    [frontends.whoami.headers.customResponseHeaders]
    Cache-Control = "public, max-age=31536000"
    [frontends.whoami.routes.test_1]
    rule = Path: /some/path
```

but I can't get that working either. Can't find any docs on if this is supposed to work or not, but I suppose it isn't.)
